tracing-attributes/src/lib.rs

```diff
@@ -442,19 +450,17 @@ impl Parse for MaybeItemFn {
 /// that takes a generic block type `B` that implements `ToTokens` (eg. `TokenStream`, `Block`).
 #[derive(Debug, Clone)]
 struct MaybeItemFnRef<'a, B: ToTokens> {
-    attrs: &'a Vec<Attribute>,
+    outer_attrs: &'a Vec<Attribute>,
+    inner_attrs: &'a Vec<Attribute>,
     vis: &'a Visibility,
     sig: &'a Signature,
     block: &'a B,
 }
 
-impl<'a> From<&'a ItemFn> for MaybeItemFnRef<'a, Box<Block>> {
-    fn from(val: &'a ItemFn) -> Self {
-        MaybeItemFnRef {
-            attrs: &val.attrs,
-            vis: &val.vis,
-            sig: &val.sig,
-            block: &val.block,
-        }
+impl TryFrom<ItemFn> for MaybeItemFn {
+    type Error = syn::Error;
+
+    fn try_from(value: ItemFn) -> Result<Self, Self::Error> {
+        syn::parse2(value.into_token_stream())
```

Comment:
i'm not sure if doing this conversion by round-tripping through the parser is really my favorite thing. looking at the `syn` docs, it looks like [the `Attribute` type has a field, `style`](https://docs.rs/syn/latest/syn/struct.Attribute.html) which [indicates whether the attribute is an inner or outer attribute](https://docs.rs/syn/latest/syn/enum.AttrStyle.html).

would it make sense to do this conversion by reusing the already-parsed `ItemFn` but separating the `Vec<Attribute>` into two `Vec`s based on the value of the `Attribute`'s `style` field, instead? that way, we don't have to re-parse the entire thing...